### PR TITLE
Update to deployer 7

### PR DESCRIPTION
**Pull request by simonrjones**
Update to use Deployer 7, ready for testing.

* See [usage documentation](https://github.com/studio24/deployer-recipes/blob/update-to-deployer-7/docs/usage.md)
* See [upgrade docs](https://github.com/studio24/deployer-recipes/blob/update-to-deployer-7/docs/upgrading.md)
* See [test branch for www.studio24.net](https://github.com/studio24/studio24/tree/feature/test-deployer-7) which uses this deployer-7 branch of deployer-recipes

@AlanJIsaacson This is ready to test to see if you think it's working reliably.

If you think it is, you can approve and merge this pull request & create a new release for v2.0.0

Please note the only thing I cannot reliably get working is [Slack notifications](https://deployer.org/docs/7.x/contrib/slack).

### CODE REVIEW

**Comment by simonrjones** [comment]
@AlanJIsaacson This is ready to test to see if you think it's working reliably.

If you think it is, you can approve and merge this pull request & create a new release for v2.0.0

Please note the only thing I cannot reliably get working is [Slack notifications](https://deployer.org/docs/7.x/contrib/slack).

* See [usage documentation](https://github.com/studio24/deployer-recipes/blob/update-to-deployer-7/docs/usage.md)
* See [upgrade docs](https://github.com/studio24/deployer-recipes/blob/update-to-deployer-7/docs/upgrading.md)
* See [test branch for www.studio24.net](https://github.com/studio24/studio24/tree/feature/test-deployer-7) which uses this deployer-7 branch of deployer-recipes

**Comment by simonrjones** [comment]
See [Slack notifications usage](https://github.com/studio24/deployer-recipes/blob/update-to-deployer-7/docs/installation.md#slack-notifications)

**Comment by simonrjones** [comment]
You can test this on the branch via:

```
composer require studio24/deployer-recipes:dev-update-to-deployer-7 --dev
```

**Review by AlanJIsaacson — APPROVED** [review]
@simonrjones All tested and double checked with others. I could do with help for the tagging for a new release eg: v2.0.0

**Review by simonrjones — COMMENTED** [review]
I've added a few comments based on a few things we previously discussed @AlanJIsaacson which I think would be good to clean up. Thanks.

File: recipe/craftcms.php [hunk 1]
```diff
@@ -0,0 +1,114 @@
...
+    $latest = run("cd $path && ls -t db-backups/ | head -n1");
+
+    writeln('Restoring Craft DB from db-backups/' . $latest);
+    output()->write(run("cd $path && php craft db/restore db-backups/$latest"));
+    writeln('DB restored');
+
+}
+);
+
+desc('Craft storage: reset storage permissions');
```
[the remark this answers is not in the document]
Comment (simonrjones) [reply recipe/craftcms.php:50]: Though if things like templates are cached then I presume these would need clearing if we're updating the site.

**Pull request merged**